js.go

```diff
@@ -1214,6 +1368,269 @@ type StreamState struct {
 	Consumers int       `json:"consumer_count"`
 }
 
+// UpdateStream updates a Stream.
+func (js *js) UpdateStream(cfg *StreamConfig) (*StreamInfo, error) {
+	if cfg == nil || cfg.Name == _EMPTY_ {
+		return nil, ErrStreamNameRequired
+	}
+
+	req, err := json.Marshal(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	usSubj := js.apiSubj(fmt.Sprintf(JSApiStreamUpdateT, cfg.Name))
+	r, err := js.nc.Request(usSubj, req, js.wait)
+	if err != nil {
+		return nil, err
+	}
+	var resp JSApiStreamInfoResponse
+	if err := json.Unmarshal(r.Data, &resp); err != nil {
+		return nil, err
+	}
+	if resp.Error != nil {
+		return nil, errors.New(resp.Error.Description)
+	}
+	return resp.StreamInfo, nil
+}
+
+// JSAPIStreamDeleteResponse is the response for a Stream delete request.
+type JSAPIStreamDeleteResponse struct {
+	APIResponse
+	Success bool `json:"success,omitempty"`
+}
+
+// DeleteStream deletes a Stream.
+func (js *js) DeleteStream(name string) error {
+	if name == _EMPTY_ {
+		return ErrStreamNameRequired
+	}
+
+	dsSubj := js.apiSubj(fmt.Sprintf(JSApiStreamDeleteT, name))
+	r, err := js.nc.Request(dsSubj, nil, js.wait)
+	if err != nil {
+		return err
+	}
+	var resp JSAPIStreamDeleteResponse
+	if err := json.Unmarshal(r.Data, &resp); err != nil {
+		return err
+	}
+	if resp.Error != nil {
+		return errors.New(resp.Error.Description)
+	}
+	return nil
+}
+
+// StreamSnapshotConfig contains options for snapshotting a Stream.
+type StreamSnapshotConfig struct {
+	// Subject to deliver the chunks to for the snapshot.
+	DeliverSubject string `json:"deliver_subject"`
+	// Do not include consumers in the snapshot.
+	NoConsumers bool `json:"no_consumers,omitempty"`
+	// Optional chunk size preference.
+	// Best to just let server select.
+	ChunkSize int `json:"chunk_size,omitempty"`
+	// Check all message's checksums prior to snapshot.
+	CheckMsgs bool `json:"jsck,omitempty"`
+}
+
+// JSAPIStreamSnapshotResponse is the response for a snapshot request.
+type JSAPIStreamSnapshotResponse struct {
+	APIResponse
+	// Estimate of number of blocks for the messages.
+	NumBlks int `json:"num_blks"`
+	// Block size limit as specified by the stream.
+	BlkSize int `json:"blk_size"`
+}
+
+// SnapshotStream creates a snapshot of a Stream. It returns a copy of the
+// snapshot data.
+func (js *js) SnapshotStream(name string, timeout time.Duration, cfg *StreamSnapshotConfig) (io.Reader, error) {
```

Comment:
It may have been fine leaving it in the config, was just asking for people view on that.

But, if we add the timeout to the API, I would add it last personally.

Reply:
I added it second so the calling code can look like this.

```go
js.SnapshotStream("foo", 5*time.Second, &nats.StreamSnapshotConfig{
	DeliverSubject: inboxSubj,
	ChunkSize:      1024,
})
```

If we add it to the end, then it'll look like this.
```go
js.SnapshotStream("foo", &nats.StreamSnapshotConfig{
	DeliverSubject: inboxSubj,
	ChunkSize:      1024,
}, 5*time.Second)
```

It's just style though. Personally, I like the way the first block looks like. But, do we have a preferred NATS style?

Reply:
Agreed that the first looks better. I was just trying to be consistent with other APIs like NextMsg() that has the timeout at the end. But let's wait anyway to see if other think that Timeout in the config itself is better.